packages/phishing-controller/src/PhishingController.ts

```diff
@@ -410,29 +480,39 @@ export class PhishingController extends BaseController<
     if (this.disabled) {
       return;
     }
+    const lastDiffTimestamp = Math.max(
+      ...this.state.phishingLists.map(({ lastUpdated }) => lastUpdated),
+    );
 
-    let hotlistDiffs;
-    try {
-      hotlistDiffs = await this.queryConfig<Hotlist>(METAMASK_HOTLIST_DIFF_URL);
-    } finally {
-      // Set `stalelistLastFetched` even for failed requests to prevent server from being overwhelmed with
-      // traffic after a network disruption.
-      this.update({
-        hotlistLastFetched: fetchTimeNow(),
+    const hotlistResponse = await this.queryConfig<DataResultWrapper<Hotlist>>(
+      `${METAMASK_HOTLIST_DIFF_URL}/${lastDiffTimestamp}`,
+    )
+      .then((d) => d)
```

Comment:
Nit: Use of `try...finally` is preferred here over `.then`/`.finally` because it allows some engines (i.e. Node.js and Chrome, still not Firefox sadly) to preserve async stack traces across these calls. If `.then` is used, the stack terminates at the `.then` obscuring any frames before that.

Reply:
Done.